chanbackup/multi.go

```diff
@@ -39,6 +40,9 @@ type Multi struct {
 	// StaticBackups is the set of single channel backups that this multi
 	// backup is comprised of.
 	StaticBackups []Single
+
+	// Encrypter encrypts backup data.
+	Encrypter lnencrypt.EncrypterDecrypter
```

Comment:
Since there are a lot of different codepaths that create `*Multi` and `*Single`, I think this is prone to a nil-pointer-dereference. It might be better in this case to create an `lnencrypt.Encrypter` every time `Encrypter` is used. If I'm being really nitpicky, this existing code should be an interface specific to the `chanbackup` package, but that is annoying since you'd need to thread through the implementing pointer in all codepaths.

Reply:
I agree, we probably shouldn't have this as a member and create the encrypter on the fly.
Perhaps we can have a function `KeyRingEncrypter()` that replaces the `GenEncryptionKey` and returns an `Encrypter` that holds its own password. Then we could also remove the encryption key from the `EncryptPayloadToWriter` and `DecryptPayloadFromReader`, given the encrypter already knows the password to use.

Reply:
Good call, that makes a lot more sense!

Reply:
Good call, that makes a lot more sense